packages/core/types/src/dml/index.ts

```diff
@@ -164,9 +169,9 @@ export type InferEntityType<T extends any> = T extends IDmlEntity<any>
   ? InferTypeOf<T>
   : T
 
-export type EntityIndex = {
+export type EntityIndex<TSchema extends DMLSchema = DMLSchema> = {
   name?: string
   unique?: boolean
-  fields: string[]
+  fields: (keyof TSchema)[]
```

Comment:
suggestion: Can we maybe use Infer<TSchema> instead? so that you can see the future generated foreign keys as well

Reply:
unless for that we handle it automatically as I suggested in slack 😄 let me know, and if they want to add some more indexes related to foreign keys then maybe they go with the .index on the relation? let me know

Reply:
resolved over huddle 💪🏻